packages/java-edition/src/binder/index.ts

```diff
@@ -10,130 +9,124 @@ import type {
+		'worldgen/surface_builder', // TODO: They are removed at some point.
 	])
 
-	const ans = new Map<
-		string,
-		{
-			category: FileCategory
-			extname: string
-			since?: ReleaseVersion
-			until?: ReleaseVersion
-		}
-	>([
-		['advancements', { category: 'advancement', extname: '.json' }],
-		['dimension', { category: 'dimension', extname: '.json', since: '1.16' }],
-		[
-			'dimension_type',
-			{ category: 'dimension_type', extname: '.json', since: '1.16' },
-		],
-		['functions', { category: 'function', extname: '.mcfunction' }],
-		[
-			'item_modifiers',
-			{ category: 'item_modifier', extname: '.json', since: '1.17' },
-		],
-		['loot_tables', { category: 'loot_table', extname: '.json' }],
-		['predicates', { category: 'predicate', extname: '.json' }],
-		['recipes', { category: 'recipe', extname: '.json' }],
-		['tags/blocks', { category: 'tag/block', extname: '.json' }],
-		['tags/entity_types', { category: 'tag/entity_type', extname: '.json' }],
-		['tags/fluids', { category: 'tag/fluid', extname: '.json' }],
-		['tags/functions', { category: 'tag/function', extname: '.json' }],
-		[
-			'tags/game_events',
-			{ category: 'tag/game_event', extname: '.json', since: '1.17' },
-		],
-		['tags/items', { category: 'tag/item', extname: '.json' }],
-		[
-			'worldgen/biome',
-			{ category: 'worldgen/biome', extname: '.json', since: '1.16' },
-		],
-		[
-			'worldgen/configured_carver',
-			{
-				category: 'worldgen/configured_carver',
-				extname: '.json',
-				since: '1.16',
-			},
-		],
-		[
-			'worldgen/configured_feature',
-			{
-				category: 'worldgen/configured_feature',
-				extname: '.json',
-				since: '1.16',
-			},
-		],
-		[
-			'worldgen/configured_structure_feature',
-			{
-				category: 'worldgen/configured_structure_feature',
-				extname: '.json',
-				since: '1.16',
-			},
-		],
-		[
-			'worldgen/configured_surface_builder',
-			{
-				category: 'worldgen/configured_surface_builder',
-				extname: '.json',
-				since: '1.16',
-				until: '1.17',
-			},
-		],
-		[
-			'worldgen/density_function',
-			{
-				category: 'worldgen/density_function',
-				extname: '.json',
-				since: '1.18',
-			},
-		],
-		[
-			'worldgen/noise',
-			{ category: 'worldgen/noise', extname: '.json', since: '1.18' },
-		],
-		[
-			'worldgen/noise_settings',
-			{
-				category: 'worldgen/noise_settings',
-				extname: '.json',
-				since: '1.16',
-			},
-		],
-		[
-			'worldgen/placed_feature',
-			{
-				category: 'worldgen/placed_feature',
-				extname: '.json',
-				since: '1.18',
-			},
-		],
-		[
-			'worldgen/processor_list',
-			{
-				category: 'worldgen/processor_list',
-				extname: '.json',
-				since: '1.16',
-			},
-		],
-		[
-			'worldgen/template_pool',
-			{
-				category: 'worldgen/template_pool',
-				extname: '.json',
-				since: '1.16',
-			},
-		],
+	const ans = new Map<string, CategoryDef>([
+		// Pre-1.21
+		dataPackResource('advancement', 'advancements', {
+			since: '1.16',
```

Comment:
This map is still missing a lot of resources (like wolf variants, damage types, trim materials, etc)